src/user.ts

```diff
@@ -26,6 +27,7 @@ export class User {
   readonly session: number;
   name?: string;
   channelId = 0;
+  userId?: number; // null userid means the user is not registered
```

Comment:
When the user is not registered, the `userId` becomes `null` or `undefined`? Please specify this in the comment or change property definition accordingly